CDS/scripts/check_date.pl

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/perl
+
+my $version='$Rev: 1 $ $LastChangedDate: 2020-05-15 16:06:33 +0100 (Fri, 15 May 2020) $';
+
+#Module to check dates are not before 2008 and set them to the current date/time if they are.
+#Expects:
+#<key>blah 	   - the name of the key to check for the date
+#END DOC
+
+use CDS::Datastore;
+use File::Slurp qw/read_file/;
+use DateTime::Format::Strptime;
+use DateTime;
+
+#START MAIN
+my $store=CDS::Datastore->new('check_date');
+
+$store->{'debug'}=$ENV{'debug'};
+
+foreach(qw/key/){
+	if(not defined $ENV{$_}){
+		print "-ERROR - you need to specify <$_> to use this module.\n";
+		exit 1;
+	}
+}
+
+my $key=$ENV{'key'};
+if($key eq ''){
+	print "-ERROR - you need to specify a value in <key> to set a metadata key.\n";
+	exit 1;
+}
+
+my @keyparts=split /:/,$ENV{'key'};
+#if no section is specified, default to "meta"
+if(scalar @keyparts==1){
+	$keyparts[1]=$ENV{'key'};
+	$keyparts[0]="meta";
+	$key="meta:$key";
+}
+
+if($keyparts[0] eq "track"){
+	splice @keyparts,1,0,'type';
+}
+
+my $existing_value=$store->get(@keyparts,undef);
+
+my $value=$existing_value;
+my $finalstring=$existing_value;
+my $format = DateTime::Format::Strptime->new( pattern => '%FT%T%z');
+my $date_time_from_datastore = $format->parse_datetime($value);
```

Comment:
what happens if the value in the key is not a valid date/time?  I think that the best behaviour would be for the script to flag it as an error and exit, unless <non-fatal/> is specified in the configuration

Reply:
I think it will most likely think the date is 1/1/1970 if there is not a valid date and output the current date/time.

Reply:
we should still flag it as an error so that we can detect (and fix) the failure rather than siltently ignore it

Reply:
I have attempted to add that.